Polyhedron/demo/Polyhedron/SMesh_type.h

```diff
@@ -0,0 +1,21 @@
+#ifndef SMESH_TYPE_H
+#define SMESH_TYPE_H
+
+#include <boost/scoped_ptr.hpp>
+#include <boost/array.hpp>
+#include <set>
```

Comment:
Why include `<boost/scoped_ptr.hpp>`, `<boost/array.hpp>`, and `<set>` whereas they are not used?